### PR TITLE
[AutoGluon] Add AutoGluon v1.1.1

### PR DESCRIPTION
**Pull request by prateekdesai04**
*GitHub Issue #, if available:*

**Note**: 
- If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

- All PR's are checked weekly for staleness. This PR will be closed if not updated in 30 days.

### Tests run

**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [ ] I have run builds/tests on commit <INSERT COMMIT ID> for my changes.

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**
<details>
<summary>Expand</summary>

- [ ] `sagemaker_remote_tests = true`
- [ ] `sagemaker_efa_tests = true`
- [ ] `sagemaker_rc_tests = true`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

</details>

### Formatting
- [ ] I have run `black -l 100` on my code (formatting tool: https://black.readthedocs.io/en/stable/getting_started.html)

#### Builds to Execute
<details>
<summary>Expand</summary>

Fill out the template and click the checkbox of the builds you'd like to execute

*Note: Replace with <X.Y> with the major.minor framework version (i.e. 2.2) you would like to start.*

- [ ] build_pytorch_training_<X.Y>_sm
- [ ] build_pytorch_training_<X.Y>_ec2

- [ ] build_pytorch_inference_<X.Y>_sm
- [ ] build_pytorch_inference_<X.Y>_ec2
- [ ] build_pytorch_inference_<X.Y>_graviton

- [ ] build_tensorflow_training_<X.Y>_sm
- [ ] build_tensorflow_training_<X.Y>_ec2

- [ ] build_tensorflow_inference_<X.Y>_sm
- [ ] build_tensorflow_inference_<X.Y>_ec2
- [ ] build_tensorflow_inference_<X.Y>_graviton
- [x] build_autogluon_inference_latest
- [x] build_autogluon_training_latest
</details>

### PR Checklist 
<details>
<summary>Expand</summary>

- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

#### NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ec2_benchmark_tests = true` or `sagemaker_benchmark_tests = true`
</details>

### Pytest Marker Checklist
<details>
<summary>Expand</summary>

- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type
</details>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: autogluon/inference/docker/1.1/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -35,7 +35,7 @@ RUN pip install --no-cache-dir -U --trusted-host pypi.org --trusted-host files.p
  && pip install --no-cache-dir -U mmdet>=3.0.0 \
  && pip install --no-cache-dir -U "mmengine" \
 # Update transformers to secure versions that fixes CVE-2023-7018
- && pip install --no-cache-dir -U setuptools>=65.5.1 \  
+ && pip install --no-cache-dir setuptools==69.5.1 \
```
Comment (tonyhoo) [inline autogluon/inference/docker/1.1/py3/Dockerfile.cpu:38]: Could you please provide a brief justification as comment for why we need to anchor the setuptools to a particular version?

File: autogluon/training/buildspec.yml [hunk 2]
```diff
@@ -28,7 +28,7 @@ images:
     image_size_baseline: 6825
```
Comment (tonyhoo) [inline autogluon/training/buildspec.yml:28]: Just double check the CPU image size stay the same

File: autogluon/training/buildspec.yml [hunk 3]
```diff
@@ -38,11 +38,11 @@ images:
   BuildAutogluonGPUTrainPy3DockerImage:
     <<: *TRAINING_REPOSITORY
     build: &AUTOGLUON_GPU_TRAINING_PY3 false
-    image_size_baseline: 26726
+    image_size_baseline: 21000
```
Comment (tonyhoo) [inline autogluon/training/buildspec.yml:41]: Great to see the image size reduction. Does that come from the base image?

File: autogluon/training/docker/1.1/py3/Dockerfile.cpu [hunk 4]
```diff
@@ -35,7 +35,7 @@ RUN pip install --no-cache-dir -U --trusted-host pypi.org --trusted-host files.p
  && pip install --no-cache-dir -U requests>=2.31.0 \
  && pip install --no-cache-dir -U pyOpenSSL>=23.2.0 \
  # Update transformers to secure versions that fixes CVE-2023-7018
- && pip install --no-cache-dir -U setuptools>=65.5.1 \
+ && pip install --no-cache-dir setuptools==69.5.1 \
```
Comment (tonyhoo) [inline autogluon/training/docker/1.1/py3/Dockerfile.cpu:38]: Same here. Comments will be helpful

File: autogluon/training/buildspec.yml (hunk 2 above)
Comment (prateekdesai04) [reply autogluon/training/buildspec.yml:28]: Yes, it roughly the same size.
Hence, keeping the baseline size similar.

File: autogluon/training/buildspec.yml (hunk 3 above)
Comment (prateekdesai04) [reply autogluon/training/buildspec.yml:41]: Yes it comes from the base image, it's smaller

File: autogluon/inference/docker/1.1/py3/Dockerfile.cpu (hunk 1 above)
Comment (prateekdesai04) [reply autogluon/inference/docker/1.1/py3/Dockerfile.cpu:38]: addressed comments

File: autogluon/training/docker/1.1/py3/Dockerfile.cpu (hunk 4 above)
Comment (prateekdesai04) [reply autogluon/training/docker/1.1/py3/Dockerfile.cpu:38]: addressed comments

**Review by tonyhoo — APPROVED** [review]

**Pull request merged**